components/engine_traits/src/sst.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::cfdefs::CfName;
+use crate::errors::Result;
+use crate::iterable::Iterable;
+use std::path::PathBuf;
+
+pub trait SstExt {
```

Comment:
`SST` maybe not a common term for engines, though I have no better idea.

Reply:
+1 I think it is an abuse of the terminology, aiui, SST is one possible implementation strategy for engines, e.g., Sled is not one, I think.

Reply:
I prefer to keep the naming identical between engines while doing the encapsulation. It makes changing the callers much easier, and it is easier to think about. Right now all the code here deals with SST because rocks deals with SST. After the encapsulation work is done we can come up with new abstractions.

Reply:
On the other hand, after this PR is done, SST features should be fully encapsulated, so we can start thinking about new abstractions. It doesn't seem like the priority to me right now, but I will note it as a work item for the future. We won't know what abstractions are necessary for other databases until we try to use them anyway.

Reply:
It might be the case that we remove the `SstExt` bound from `KvEngine` in the future, making it an optional feature for engines to implement; or we might have a generic "import-some-external-dumpfile" feature.

Reply:
I am willing to rename this trait though if you like.

Reply:
:+1:

Reply:
looks fine for me~